### ISSUE TITLE
Adding number of horizontal bracing members to mast_section_geometry table

### ISSUE THREAD

**Issue by kersting** [IEA-Task-43/digital_wra_data_standard#174]
We currently have the field `lattice_number_of_diagonal_bracing_members` that helps one to calculate solidity which is necessary to calculate Ct according to IEC 614000-12. I'd like to propose a similar field for the horizontal bracing members called 'lattice_number_of_horizontal_bracing_members'. The reason for that is that we have so far assumed that towers would only have 2 horizontal members in the section; i.e. one on the bottom and one on the top; however, I am showing below a tower with multiple horizontal members in a section. Those need to be counted in order to properly calculate the solidity of the tower.

![image](https://user-images.githubusercontent.com/19154103/169404265-98e0d1ef-00ed-4982-9c14-051e0f8e043d.png)

**Comment by stephenholleran** [comment]
@kersting don't we just calculate the solidity of the repetitive section i.e. what I have highlighted in the red box.
![image](https://user-images.githubusercontent.com/25622575/169558080-6bbb258c-e22a-4979-be97-32c07c831df9.png)

So we only need to account for a single horizontal member in that little section.

**Comment by kersting** [comment]
@stephenholleran yes but the problem that we're facing now is that a tower may have variable patterns within the section (see the screenshot below). For the screenshot below that would work only if one accounts for the repetitive patterns in the interface with the next section. What if the point of interest was away from the intersection with the other mast section? 

This brings an additional challenge and I think that the number of horizontal bracing members is just one piece of the puzzle. 

One way of dealing with this issue is to redefine the field `lattice_has_horizontal_member`. This field could be related to the interface connecting both sections. Perhaps even renaming that variable to `lattice_has_horizontal_section_connector`- and adding another field to account for the number of lattice horizontal bracing members that are not related to the interface between 2 sections. 

For my first example, the first variable would be true (so you'd have 2 interfaces) and the number of horizontal members would be 4. For my second example, the first variable would be true (so you'd have 2 interfaces) and the number of horizontal members would be 0.

![image](https://user-images.githubusercontent.com/19154103/169567947-c534cd3c-b5fc-45a2-897e-c0f810522a36.png)

**Comment by abohara** [comment]
@kersting I don't technically have anything to add. This is a tangential question.  I generally see a limited number of OEM / models in our markets and so once compliance for that model is established each one does not required re-validation. I do wonder if there is truss pattern that the IEC standard hand in mind. The wider world of truss's have a lot more variable to them (and dedicate software for it). Should we be clear about supporting some common patterns and then for unusual patterns , we could have a field that says 'solidity' that reports it from external calculations ? In areas, where they may be more models or custom towers out, we would not be able to support it with limited number of variables ( or the number of fields will get so large than it will be hard to understand how to use them for common models). Thoughts @kersting @stephenholleran

**Comment by kersting** [comment]
@abohara and @stephenholleran the number of truss patterns is not that many based on the tower diagrams that we gathered for years internally from many countries. I think the reason why many questions around this subject have been popping up is that as we develop this internally, we see the missing parts that we didn't capture on version 1 of the standard. I think we have exhausted the number of possible problems that could arise when calculating solidity. If we get to the point that the energy to add additional fields vs the gains becomes too much, then I agree that we should add a solidity field even though I am opposed to the idea of adding calculated fields. I still hope that after we align on these lose ends, that we have an open path to perform all the IEC checks for any given tower in a robust way which will bring efficiency to everyone in the industry.

**Comment by kersting** [comment]
@stephenholleran thanks for the feedback. I'm thinking about this being an issue if one is performing a power performance test and we end up underestimating the solidity of an existing tower. Perhaps one way of dealing with additional members in a tower section is to resuscitate the idea of adding the area of additional members in the section as proposed here https://github.com/IEA-Task-43/digital_wra_data_standard/issues/154. This would eliminate the need for additional description and we could set that variable as optional.

**Comment by stephenholleran** [comment]
Hi @kersting,

If it is just power performance testing that is the remaining concern, does the mast section geometry apply to the top mounted anemometers as they are mounted above the mast and have other conditions to meet?

**Comment by kersting** [comment]
@stephenholleran 61400-12 is a standard for power performance test but used for resource assessment as well. The standard recommends that the disturbance from the boom mounted anemometers is set below 1%. This is related to the solidity- and Ct of the tower. If one is using boom mounted anemometers for shear analysis during a power performance test and the disturbance is higher than 1%, the user will be penalized with higher uncertainty by some consultants.

**Comment by stephenholleran** [comment]
>  If one is using boom mounted anemometers for shear analysis during a power performance test and the disturbance is higher than 1%, the user will be penalized with higher uncertainty by some consultants.

I suppose that is what I am asking. Are any other anemometers mounted down the side of the mast used for power performance testing? I mistakenly assumed that it is just the top mounted ones.

**Comment by kersting** [comment]
@stephenholleran yes, shear is used in the latest IEC standard and it is calculated side-mounted anemometers if a remote sensing device is not available.
